tests/test_conversion.py

```diff
@@ -0,0 +1,238 @@
+import os
+import sys
+from contextlib import ContextDecorator
+from functools import wraps
+from unittest.mock import patch
+
+import deepspeed
+import torch
+torch.backends.cudnn.deterministic = True
+from transformers import GPT2LMHeadModel, GPT2Tokenizer
+
+from megatron import get_args, get_tokenizer, global_vars, initialize_megatron
+from megatron.testing_utils import (
+    TestCasePlus,
+    mockenv_context,
+    set_seed,
+    torch_assert_equal,
+    torch_assert_close
+)
+from megatron.training import setup_model_and_optimizer
+from megatron.checkpointing import load_checkpoint, save_checkpoint
+
+from pretrain_gpt import get_batch_pipe, model_provider
+from pretrain_gpt import model_provider as gpt_model_provider, get_batch_pipe as get_gpt_batch_pipe
+import json
+
+# fix the relative path
+sys.path.append('tools/convert_checkpoint/')
+from tools.convert_checkpoint import deepspeed_to_transformers, deepspeed_to_megatron
+from test_model import flatten_arguments, get_default_args
+
+
+def reset_global_variables():
+    global_vars._GLOBAL_ARGS = None
+    global_vars._GLOBAL_NUM_MICROBATCHES_CALCULATOR = None
+    global_vars._GLOBAL_TOKENIZER = None
+    global_vars._GLOBAL_TENSORBOARD_WRITER = None
+    global_vars._GLOBAL_ADLR_AUTORESUME = None
+    global_vars._GLOBAL_TIMERS = None
+
+
+class TestConversion(TestCasePlus):
+    def setUp(self) -> None:
+        super().setUp()
+        set_seed()
+        reset_global_variables()
+
+        self.fp16 = False
+        self.dist_env_1_gpu = dict(
+            MASTER_ADDR="localhost", MASTER_PORT="9994", RANK="0", LOCAL_RANK="0", WORLD_SIZE="1"
+        )
+
+    def save_and_get_ds_megatron_output(self, args, ds_megatron_model, ds_megatron_token, ds_megatron_dir):
+        # args for checkpointing
+        args.save, args.no_save_optim = ds_megatron_dir, True
+        # save deepspeed megatron model
+        save_checkpoint(1, ds_megatron_model, None, None)
+
+        ds_megatron_model = ds_megatron_model[0]
+
+        # get processed batch
+        _, _, attention_mask = ds_megatron_token
+
+        # disable activation checkpoint
+        ds_megatron_model.module.activation_checkpoint_interval = 0
+        # resemble _exec_schedule in /deepspeed/runtine/pipe/engine.py
+        ds_megatron_model._compute_loss = False
+        ds_megatron_model.fwd_outputs = []
+
+        ds_megatron_model.eval()
+        # due to the hack in https://github.com/bigscience-workshop/Megatron-DeepSpeed/blob/68b46f201aad8803d0699f76b100663553e89e1b/pretrain_gpt.py#L74
+        args.attn_mask = attention_mask
+
+        # use deepspeed pipeline thing 
+        ds_megatron_model.pipe_buffers["inputs"].append( ds_megatron_token )
+        ds_megatron_model.pipe_buffers["outputs"].append( None )
```

Comment:
```suggestion
        ds_megatron_model.pipe_buffers["inputs"].append(ds_megatron_token)
        ds_megatron_model.pipe_buffers["outputs"].append(None)
```

We aren't using any auto-formatters, so please let's try to keep the code consistent voluntarily

Reply:
I'll run `black` and `isort` before the final merge.

Reply:
The problem is that we can't do auto-formatters for the main code since megatron doesn't do it and if we reformat things it'll be very difficult to sync between code bases.

I guess nothing stops us from using autoformatters for the test suite since there is no overlap there.

If we do that please use the same settings as HF transformers and add a target to `Makefile` to activate it, probably like:
```
make style
```
so for now it'd only run for `tests`.

If you run into difficulties please let me know.

Reply:
Here is a suggestion: Let's do that in another PR, and merge this one as soon as it's ready.